motoman_hc20_support/urdf/hc30pl_macro.xacro

```diff
@@ -0,0 +1,163 @@
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20/collision/link_4_r.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_5_b">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20dtp/visual/link_5_b.dae"/>
+        </geometry>
+        <origin rpy="0 1.57075 0"/>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20dtp/collision/link_5_b.stl"/>
+        </geometry>
+        <origin rpy="0 1.57075 0"/>
+      </collision>
+    </link>
+    <link name="${prefix}link_6_t">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20dtp/visual/link_6_t.dae"/>
+        </geometry>
+        <origin rpy="0 1.57075 0"/>
+        <xacro:material_yaskawa_white/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_hc20_support/meshes/hc20dtp/collision/link_6_t.stl"/>
+        </geometry>
+        <origin rpy="0 1.57075 0"/>
+      </collision>
+    </link>
+    <link name="${prefix}tool0"/>
+    <!-- end of link list -->
+
+    <!-- joint list -->
+    <joint name="${prefix}joint_1_s" type="revolute">
+      <parent link="${prefix}base_link"/>
+      <child link="${prefix}link_1_s"/>
+      <origin xyz="0 0 0.380" rpy="0 0 0" />
+      <axis xyz="0 0 1" />
+      <limit lower="${radians(-210)}" upper="${radians(210)}" effort="917.28" velocity="${radians(80)}"/>
+    </joint>
+    <joint name="${prefix}joint_2_l" type="revolute">
+      <parent link="${prefix}link_1_s"/>
+      <child link="${prefix}link_2_l"/>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <axis xyz="0 1 0" />
+      <limit lower="${radians(-154)}" upper="${radians(180)}" effort="1223.04" velocity="${radians(80)}"/>
+    </joint>
+    <joint name="${prefix}joint_3_u" type="revolute">
+      <parent link="${prefix}link_2_l"/>
+      <child link="${prefix}link_3_u"/>
+      <origin xyz="0 0 0.820" rpy="0 0 0" />
+      <axis xyz="0 -1 0" />
+      <limit lower="${radians(-67)}" upper="${radians(247)}" effort="642.88" velocity="${radians(120)}"/>
+    </joint>
+    <joint name="${prefix}joint_4_r" type="revolute">
+      <parent link="${prefix}link_3_u"/>
+      <child link="${prefix}link_4_r"/>
+      <origin xyz="0.880 0 0" rpy="0 0 0" />
+      <axis xyz="-1 0 0" />
+      <limit lower="${radians(-15)}" upper="${radians(15)}" effort="368.48" velocity="${radians(112)}"/>
+    </joint>
+    <joint name="${prefix}joint_5_b" type="revolute">
+      <parent link="${prefix}link_4_r"/>
+      <child link="${prefix}link_5_b"/>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <axis xyz="0 -1 0" />
+      <limit lower="${radians(-15)}" upper="${radians(15)}" effort="141.12" velocity="${radians(132)}"/>
```

Comment:
same here

Reply:
The HC30PL is an HC20 for Palletizing, so the wrist is always kept down. This is what allows to manipulate 30 kg instead of 20 kg. But I'm realizing that the 15 deg in the manual is relative to the ground, not the previous link. The B-axis range probably needs to be reviewed to allow more motion. I don't think there is any way to put at restriction on the B-axis by considering the other axes current position.

Reply:
> I don't think there is any way in a URDF to put at motion restriction on the B-axis relative to the ground or by considering the other axes current position.

no, there currently isn't. See my previous comment.

Reply:
'the wrist' means both `joint_4_r` and `joint_5_b`, correct?